File: tests/fixtures/py_calc/calc.py
"""Tiny calculator used as a harness fixture. Contains one deliberate bug."""


def add(a: float, b: float) -> float:
    return a + b


def divide(a: float, b: float) -> float:
    if b == 0:
        raise ZeroDivisionError("cannot divide by zero")
    # BUG: integer division truncates; tests expect true division.
    return a // b


def slugify(text: str) -> str:
    return "-".join(text.lower().split())

File: tests/fixtures/ts_calc/calc.ts
export function add(a: number, b: number): number {
  return a + b;
}

export function divide(a: number, b: number): number {
  if (b === 0) throw new Error("cannot divide by zero");
  // BUG: truncates; tests expect true division.
  return Math.trunc(a / b);
}

export function slugify(text: string): string {
  return text.toLowerCase().split(/\s+/).join("-");
}
